### PR TITLE
Allow multiple nics for controller and compute nodes

### PR DESCRIPTION
**Pull request by toabctl**
This is useful to test real world setups where nodes do have multiple
nics.
Setting the new variable $nics (defaults to 1) generates $nics network
interfaces per node.

### CODE REVIEW

File: scripts/lib/mkcloud-libvirt.sh [hunk 1]
```diff
@@ -397,5 +397,7 @@ function libvirt_do_shutdowncloud()
 
 function libvirt_do_macfunc
 {
-    printf "$macprefix:77:77:%02x" $1
+    local nodenumber=$1
+    local nicnumber=${2:-"1"}
+    printf "$macprefix:77:%02x:%02x" $nicnumber $nodenumber
```
Comment (bmwiedemann) [inline scripts/lib/mkcloud-libvirt.sh]: maybe use $((118+nicnumber)) to be backwards compatible

**Comment by toabctl** [comment]
> there is a bug somewhere - travis shows that the generated admin xml contains ~16 NICs where each one has a single-letter MAC-addr. Probably some string is mistaken for an array in python.

Seems to be a test bug. Hopefully fixed

File: scripts/lib/mkcloud-libvirt.sh (hunk 1 above)
Comment (toabctl) [inline scripts/lib/mkcloud-libvirt.sh]: Why backwards compatible? I this this is more intuitive and easier to understand.

File: scripts/mkcloud [hunk 2]
```diff
@@ -582,9 +583,12 @@ function setupnodes
                 drbdnode_mac_vol="${drbdnode_mac_vol#+}"
             fi
         fi
-
+        mac_params=""
```
Comment (toabctl) [inline scripts/mkcloud]: Done for ```mac_params``` but why do I need it for the loop var ?

File: scripts/mkcloud [hunk 3]
```diff
@@ -1027,6 +1031,8 @@ Optional
         of $nodenumber - $nodenumbercompute are used as HA cluster nodes
     vcpus=1         (default $vcpus)
         set the number of CPU cores per compute node
+    nics=1         (default $nics)
+        set the number of network interfaces
```
Comment (toabctl) [inline scripts/mkcloud]: Done

**Pull request merged**